src/snowflake/snowpark/mock/functions.py

```diff
@@ -525,9 +525,11 @@ def mock_to_timestamp(
             if data is None:
                 res.append(None)
                 continue
-            if auto_detect and data.isnumeric():
+            if auto_detect and (
+                isinstance(data, int) or (isinstance(data, str) and data.isnumeric())
+            ):
                 res.append(
-                    datetime.datetime.utcfromtimestamp(process_numeric_time(data))
+                    datetime.datetime.utcfromtimestamp(process_numeric_time(int(data)))
```

Comment:
We don't have to cast to `int` here, `process_numeric_time` is expecting a `str` and will cast that to `int`.

Reply:
updated